src/Illuminate/Database/Eloquent/Factory/RelationRequest.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Factory;
+
+use Closure;
+use BadMethodCallException;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Database\Eloquent\Relations\Relation;
+
+class RelationRequest
+{
+    use PrototypesModels;
+
+    /**
+     * The parent model requesting relations.
+     *
+     * @var \Illuminate\Database\Eloquent\Model
+     */
+    protected $class;
+
+    /**
+     * The batch number.
+     *
+     * @var int
+     */
+    protected $batch;
+
+    /**
+     * The StateManager that holds all definitions.
+     *
+     * @var \Illuminate\Database\Eloquent\Factory\StateManager
+     */
+    protected $stateManager;
+
+    /**
+     * A cached name of the related model class.
+     *
+     * @var string|null
+     */
+    protected $cachedRelatedClass;
+
+    /**
+     * The (possibly nested) relations path.
+     *
+     * @var string
+     */
+    public $path;
+
+    /**
+     * The build function.
+     *
+     * @var callable|null
+     */
+    public $builder = null;
+
+    /**
+     * Create a new relationship request.
+     *
+     * @param  string  $class
+     * @param  int  $batch
+     * @param  \Illuminate\Database\Eloquent\Factory\StateManager  $stateManager
+     * @param  mixed  $args
+     * @return void
+     */
+    public function __construct($class, $batch, StateManager $stateManager, $args)
+    {
+        [$this->class, $this->batch, $this->stateManager] = [$class, $batch, $stateManager];
+
+        collect($args)
+            ->pipe(Closure::fromCallable([$this, 'findAndPopRelationName']))
+            ->tap(Closure::fromCallable([$this, 'failOnMissingRelation']))
+            ->each(Closure::fromCallable([$this, 'parseArgument']));
+    }
+
+    /**
+     * Create a new relationship request for nested relations.
+     *
+     * @return static
+     */
+    public function createNestedRequest()
+    {
+        $request = new static($this->getRelatedClass(), $this->batch, $this->stateManager, $this->getNestedPath());
+        $request->amount = $this->amount;
+        $request->attributes = $this->attributes;
+        $request->builder = $this->builder;
+        $request->states = $this->states;
+
+        return $request;
+    }
+
+    /**
+     * Get the current batch no.
+     *
+     * @return int
+     */
+    public function getBatch()
+    {
+        return $this->batch;
+    }
+
+    /**
+     * Get the nested path beyond immediate relation.
+     *
+     * @param  string|null  $path
+     * @return string
+     */
+    public function getNestedPath($path = null)
+    {
+        $nested = explode('.', $path ?: $this->path);
+
+        array_shift($nested);
+
+        return implode('.', $nested);
```

Comment:
You can omit the two first line and accept this one (In `getRelationName` function you use the same logic)

```suggestion
        return implode('.', $this->getRelationName($path));
```